### PR TITLE
Expose prow config in JSON

### PR DESCRIPTION
**Pull request by kargakis**
It will help configure branding in prow.

/cc @BenTheElder @cjwagner

### CODE REVIEW

File: prow/cmd/deck/main.go [hunk 1]
```diff
@@ -373,7 +373,7 @@ func handleConfig(ca configAgent) http.HandlerFunc {
 		// TODO(bentheelder): add the ability to query for portions of the config?
 		w.Header().Set("Cache-Control", "no-cache")
 		config := ca.Config()
-		b, err := yaml.Marshal(config)
+		b, err := json.Marshal(config)
```
Comment (BenTheElder) [inline prow/cmd/deck/main.go]: I'd still like a way to expose the yaml so we can easily compare the config file currently loaded to the config stored elsewhere.

**Comment by BenTheElder** [comment]
> It will help configure branding in prow.

Are we going to lazy-load the branding?

File: prow/cmd/deck/main.go (hunk 1 above)
Comment (kargakis) [reply prow/cmd/deck/main.go]: Added a separate endpoint for JSON - let me know if that's ok with you.

**Comment by kargakis** [comment]
> Are we going to lazy-load the branding?

This is the only way I could think of. Are there any alternatives?

File: prow/cmd/deck/main.go (hunk 1 above)
Comment (BenTheElder) [reply prow/cmd/deck/main.go]: thanks!

**Comment by BenTheElder** [comment]
> This is the only way I could think of. Are there any alternatives?

There are, not ones I'm particularly though and I wasn't sure if this was OK for you all.

We could template the files and inject the customization Deck side (gross, adds a lot more complexity / serving cost just to swap some colors), or we could remove project specific things (logos) from most pages other than colors and just use the stylesheet.

The only downside I see to this is there might be slightly more load time client side, but I think if it's written well it won't be noticeable and these pages render with JS anyhow.

**Comment by BenTheElder** [comment]
Actually @kargakis we should make an endpoint that serves just a specific portion of the config, IIRC this will serve *all* of the prow config including all of the jobs which will be far too large for the k8s deck to be serving to every page load.

File: prow/cmd/deck/main.go [hunk 2]
```diff
@@ -94,6 +94,7 @@ func main() {
 	mux.Handle("/log", gziphandler.GzipHandler(handleLog(ja)))
 	mux.Handle("/rerun", gziphandler.GzipHandler(handleRerun(kc)))
 	mux.Handle("/config", gziphandler.GzipHandler(handleConfig(configAgent)))
+	mux.Handle("/config.js", gziphandler.GzipHandler(handleJSONConfig(configAgent)))
```
Comment (BenTheElder) [inline prow/cmd/deck/main.go]: we should probably make an endpoint to serve just the deck config, [the existing endpoint](https://prow.k8s.io/config) serves 777KB of prow config on our deployment ATM.

Comment (cjwagner) [reply prow/cmd/deck/main.go]: :this_is_fine:

Comment (kargakis) [reply prow/cmd/deck/main.go]: updated to include only the branding config.

File: prow/cmd/deck/runlocal [hunk 3]
```diff
@@ -17,4 +17,5 @@ cd static
 curl "https://prow.k8s.io/data.js?var=allBuilds" > data.js
 curl "https://prow.k8s.io/tide.js?var=tideData" > tide.js
 curl "https://prow.k8s.io/plugin-help.js?var=allHelp" > plugin-help.js
+curl "https://prow.k8s.io/branding.js?var=config" > branding.js
```
Comment (cjwagner) [inline prow/cmd/deck/runlocal]: Should we change the variable name too now that it is just the branding, not the full config?
LGTM though!

Comment (kargakis) [reply prow/cmd/deck/runlocal]: done

**Pull request merged**